src/Engine/MapController.java

```diff
@@ -575,6 +548,10 @@ public void animationEnded(GameEvent event, boolean animEventQueueIsEmpty)
       }
       else
       {
+        if( event.shouldEndTurn() ) {
+          handleEndTurn();
+        }
```

Comment:
I'm not necessarily opposed to having an event for the end of a turn, but adding a new base-class hook that is by definition used in only one subclass would cause me to question my approach.

Reply:
Also, why move it into an event? Especially since the event doesn't actually *do* anything, which is what events were created to do.

Reply:
I feel like control flow is a bit of a special case all over the place, being the return values of functions and stuff. It seemed not too egregious here.
Swap actions are a thing, yes, and it also enables listening for end-of-turn.
I could have done swap actions without making an event, but I think it would have been messy without some sort of rework.

...It might not be such a bad idea to move the turn-end logic directly into the turn-end events, which could deduplicate code and remove the need for this hook. Hmm.